typescript/cli/src/deploy/core.ts

```diff
@@ -461,3 +469,77 @@ async function writeAgentConfig(
   writeJson(outPath, agentConfig);
   logGreen('Agent configs written');
 }
+
+interface DeployParams {
+  context: WriteCommandContext;
+  chain: ChainName;
+  config: CoreConfig;
+}
+/**
+ * Executes the core deploy command.
+ */
+export async function runCoreDeploy({
+  context,
+  chain,
+  config,
+}: {
+  context: WriteCommandContext;
+  chain: ChainName;
+  config: CoreConfig;
+}) {
+  const {
+    signer,
+    isDryRun,
+    chainMetadata,
+    dryRunChain,
+    registry,
+    skipConfirmation,
+  } = context;
+
+  // Select a dry-run chain if it's not supplied
+  if (dryRunChain) {
+    chain = dryRunChain;
+  } else if (!chain) {
+    if (skipConfirmation) throw new Error('No chain provided');
+    chain = await runSingleChainSelectionStep(
+      chainMetadata,
+      'Select chain to connect:',
+    );
+  }
+  const deploymentParams: DeployParams = {
+    context,
+    chain,
+    config,
+  };
+
+  await runDeployPlanStep(deploymentParams);
+  await runPreflightChecksForChains({
+    ...deploymentParams,
+    chains: [chain],
+    minGas: MINIMUM_CORE_DEPLOY_GAS,
+  });
+
+  const userAddress = await signer.getAddress();
+
+  const initialBalances = await prepareDeploy(context, userAddress, [chain]);
+
+  logBlue('All systems ready, captain! Beginning deployment...');
+  const evmCoreModule = await EvmCoreModule.create({
+    chain,
+    config,
+    multiProvider: context.multiProvider,
+  });
+
+  await completeDeploy(context, 'core', initialBalances, userAddress, [chain]);
+  const deployedAddresses = evmCoreModule.serialize();
```

Comment:
im not sure exactly everything going on in these preflight checks, prepare, and complete but wondering if these all make sense still

Reply:
We can definitely remove and consolidate some of the logic here.
- preflight has gas balance checker, checks that protocol is ethereum
- prepare and complete has gas used calculation logic, reset fork for dry run

For the sake of time, i'll leave this out of the PR for now